File: Army(C++)/Warlock.cpp
#include "Warlock.hpp"

Warlock::Warlock(const std::string& name): Wizard(name) {
    this->spellBook->insert(std::pair<SpellList, Spell&>(AVALANCHE, *new MagicAttack(35, 18)));
    this->summonManaConsume = 100;
}

Deamon& Warlock::summon() const {
    if ( this->getMana().getCurrentMana() >= this->summonManaConsume ) {

        this->getMana().takeMana(this->summonManaConsume);
        return *Deamon::getDeamon();
    }
    throw OutOfMana();
}

File: Army(java)/Warlock.java
public class Warlock extends Wizard {
    private Demon demon;
    private int summonManaConsume = 100;

    public Warlock(String name) {
        super(name);
        this.spellBook.put(SpellList.AVALANCHE, new MagicAttack(35, 15));
    }
    public Demon summon(String name) {
        try {
            this.getMana().takeMana(summonManaConsume);
        } catch ( OutOfManaException e ) {
            return null;
        }
        return Demon.getDemon(name);
    }
}